### DISCUSSION TITLE
Is there a way to keep VAE/Text Encoder preferences for each UI when changing between them?

### DISCUSSION THREAD

**Discussion by MigCurto** [lllyasviel/stable-diffusion-webui-forge#1875] [Q&A]
Would be great to keep different VAE settings checked for each of the UI selection when changing between XL and FLUX , is there already a way to accomplish that?

tks.

**Comment by zeittresor** [comment]
hi, i am not sure about this but it might work with the "config-presets" extension.. just by selecting different VAE/TE, saving that as a new config named like the mode for example "VAE-TE for SD" or "VAE-TE for Flux".. (on the right side below the preview window) using the extension. If you change the mode (sd, sdxl, flux, what ever...) just select the matching configuration.. but i have not tried this, not sure it is working or not..

Update: It seams not to work so far... (the config says: # Note that components on the top row of the UI cannot be added here, such as "setting_sd_model_checkpoint", "setting_sd_vae", and "setting_CLIP_stop_at_last_layers".) :-/

**Comment by altoiddealer** [comment]
Just mentioning that I do have something similar reported in this Issues here (Header "Checkpoint Preferences")

Not exactly the same thing but if this existing feature was corrected/updated, this could handle what you say - albeit each model would have to be saved with desired modules.

https://github.com/lllyasviel/stable-diffusion-webui-forge/issues/1532

**Reply by MigCurto** [reply]
Nice, for me at least is just an inconvenience , but I recon it would make a nice addition if at least changing Flux and XL spaces would retain VAE settings (like it does for other settings)
